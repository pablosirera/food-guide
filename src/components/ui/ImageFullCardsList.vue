<script setup>
import ImageFullCard from '@/components/ui/ImageFullCard.vue'
import { ref } from 'vue'

defineProps({
  items: {
    type: Array,
    default: () => [],
  },
  size: {
    type: String,
    default: 'normal',
  },
  hasStyleSelected: {
    type: Boolean,
    default: false,
  },
})
const emit = defineEmits(['select-item'])

const itemSelectedId = ref(null)

const checkIfIsSelected = itemId => {
  return itemSelectedId.value === itemId
}
const selectItem = item => {
  itemSelectedId.value = item
  emit('select-item', item)
}
</script>

<template>
  <section class="flex flex-wrap">
    <slot name="first" />
    <ImageFullCard
      v-for="(item, index) in items"
      :key="index"
      :item="item"
      :size="size"
      :has-style-selected="hasStyleSelected"
      :selected="checkIfIsSelected(item.id)"
      @click="selectItem(item.id)"
    />
  </section>
</template>
