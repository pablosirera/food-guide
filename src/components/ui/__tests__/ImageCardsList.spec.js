import { mount } from '@vue/test-utils'
import ImageFullCardsList from '@/components/ui/ImageFullCardsList.vue'
import { describe, expect, it } from 'vitest'

describe('Components > UI > ImageFullCardsList', () => {
  it('renders', () => {
    const wrapper = mount(ImageFullCardsList)
    expect(wrapper.exists).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
