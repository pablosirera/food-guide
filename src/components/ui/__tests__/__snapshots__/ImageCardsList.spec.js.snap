// Vitest Snapshot v1

exports[`Components > UI > ImageCardsList > renders 1`] = `
"<section class=\\"flex flex-wrap justify-between sm:justify-center\\">
  <!--v-if-->
</section>"
`;
